domain/modelconfig/validators/validators.go

```diff
@@ -165,20 +168,11 @@ func LoggingTracePermissionChecker(canTrace bool) config.ValidatorFunc {
 	}
 }
 
-// SecretBackendProvider is responsible for checking if a given secrets backend
-// exists.
-type SecretBackendProvider interface {
-	// HasSecretsBackend checks if the provided secrets backend name exists. If
-	// an error occurs during checking for the backend false and a subsequent
-	// error is returned.
-	HasSecretsBackend(string) (bool, error)
-}
-
 // SecretBackendChecker is responsible for asserting the secret backend in the
 // updated model config is a valid secret backend in the controller. If the
 // secret backend has not changed or is the default backend then no validation
 // is performed. Any validation errors will satisfy config.ValidationError.
-func SecretBackendChecker(provider SecretBackendProvider) config.ValidatorFunc {
+func SecretBackendChecker(modelType coremodel.ModelType) config.ValidatorFunc {
```

Comment:
Looking at this more I think the concept of validators are fundamentally dead. I think all of this can be shifted into handlers OnSave etc. We get messy with transactions across boundaries having them separated.

For the validators we have they can be moved to handlers I think.

Reply:
Not all validators. Some validators are just for the scalar values passed in, eg logging config. So we still need validators for those cases.
For secret backends, yes, we are split. We could just move all the validation stuff for that to the handler. The fact that we are "validating" even in that case is why I originally had the handler return a ValidationError as to my mind that was the correct semantic meaning of that error.

Reply:
All secret backend validation now done in the handler